modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotCheckProcess.java

```diff
@@ -369,19 +381,9 @@ private void reducePreparationAndMetasCheck(
         }
     }
 
-    /** Finds current snapshot name from the metas. */
-    private @Nullable String snpName(Map<UUID, SnapshotCheckResponse> results) {
-        for (SnapshotCheckResponse nodeRes : results.values()) {
-            if (nodeRes == null || F.isEmpty(nodeRes.metas))
-                continue;
-
-            assert nodeRes.metas.get(0) != null : "Empty snapshot metadata in the results";
-            assert !F.isEmpty(nodeRes.metas.get(0).snapshotName()) : "Empty snapshot name in a snapshot metadata.";
-
-            return nodeRes.metas.get(0).snapshotName();
-        }
-
-        return null;
+    /** @return Unique operation context id depending on request type and snapshot name. */
+    private static String contextId(SnapshotCheckProcessRequest req) {
```

Comment:
Let's forbid concurrent checking of full and incremental index for a while. Let's discuss it in separate ticket

Reply:
Ok. But that woul decline concurrent: restoring snapshot and checking incremental, checking different increments of the same snapshot, quick check and full check